File: corpus/fixtures/python/sample.py
def hello(name: str) -> str:
    return "hello " + name

class Greeter:
    def greet(self, name: str) -> str: ...

class EnglishGreeter(Greeter):
    def greet(self, name: str) -> str:
        return hello(name)

class SpanishGreeter(Greeter):
    def greet(self, name: str) -> str:
        return "hola " + name

def chain_a(name: str) -> str: return chain_b(name)
def chain_b(name: str) -> str: return chain_c(name)
def chain_c(name: str) -> str: return hello(name)

def source() -> str: return user_input()
def user_input() -> str: return "user"
def sink(v: str) -> None: pass

def taint_flow() -> None:
    sink(source())

def clone_pair_a(x: int, y: int) -> int: return x + y + 1
def clone_pair_b(x: int, y: int) -> int: return x + y + 1

File: corpus/fixtures/kotlin/sample.kt
fun hello(name: String): String = "hello $name"

interface Greeter {
    fun greet(name: String): String
}

class EnglishGreeter : Greeter {
    override fun greet(name: String): String = hello(name)
}

class SpanishGreeter : Greeter {
    override fun greet(name: String): String = "hola $name"
}

fun chainA(name: String): String = chainB(name)
fun chainB(name: String): String = chainC(name)
fun chainC(name: String): String = hello(name)

fun source(): String = userInput()
fun userInput(): String = "user"
fun sink(v: String) {}

fun taintFlow() {
    sink(source())
}

fun clonePairA(x: Int, y: Int): Int = x + y + 1
fun clonePairB(x: Int, y: Int): Int = x + y + 1
